src/Api/Controllers/OrganizationUsersController.cs

```diff
@@ -270,8 +270,17 @@ public async Task PutGroups(string orgId, string id, [FromBody] OrganizationUser
         [HttpPut("{userId}/reset-password-enrollment")]
         public async Task PutResetPasswordEnrollment(string orgId, string userId, [FromBody] OrganizationUserResetPasswordEnrollmentRequestModel model)
         {
+            var user = await _userService.GetUserByIdAsync(new Guid(userId));
+            var authenticated = await _userService.CheckPasswordAsync(user, model.MasterPasswordHash);
```

Comment:
And use callingUser here

Reply:
I'm not sure I follow - there appears to be separate users - the user that's enrolling and the user doing the enrolling. It seems like they can be the same by accident, but the intent seems to allow them to vary. Does using the `User` principle bind these together incorrectly or am I misreading something?

Reply:
In the case of the enrollment, the calling user is always the same as the user being enrolled. Also why we are also asking for their master password, which only they would know.

Reply:
@kspearrin great question! After reviewing the issue, the only exploitable vulnerability here is if a user walked away from their machine with their vault unlocked, someone could walk by and enroll them in password reset as current the app does not prompt for the master password.

The only time an admin can enroll a user in password reset is when they are invited to an org. After that, the only way to enroll is if the user self-enrolls so I added the reprompt. Does that make sense?

Reply:
@kspearrin PR has been updated - thanks for the feedback!